components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -887,6 +887,19 @@ void BatLedgerImpl::DisconnectWallet(
                 _1));
 }
 
+void BatLedgerImpl::DisconnectExternalWallets(
+    DisconnectExternalWalletsCallback callback) {
+  auto weak_self = AsWeakPtr();
+  auto shared_callback =
+      std::make_shared<DisconnectExternalWalletsCallback>(std::move(callback));
+
+  ledger_->DisconnectExternalWallets(
```

Comment:
We can `std::move<>()` `shared_callback` into the lambda, which doesn't require (atomic) reference count manipulation.
Or even:
```
void BatLedgerImpl::DisconnectExternalWallets(
    DisconnectExternalWalletsCallback callback) {
  ledger_->DisconnectExternalWallets(
      [weak_self = AsWeakPtr(),
       callback = std::make_shared<DisconnectExternalWalletsCallback>(
           std::move(callback))](ledger::type::Result result) {
        if (weak_self)
          std::move(*callback).Run(result);
      });
}
```